### PR TITLE
Adding ablity to search for methods and constructors

### LINKED ISSUE #11: method call,

### LINKED ISSUE #12: constructor call,

### PR DESCRIPTION
**Pull request by shawn-hurley**
fixes #11 
fixes #12 

Also adds a dockerfile to build the base image. A follow up PR to add github actions here to build the image and push to quay will be added.

### CODE REVIEW

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java [hunk 1]
```diff
@@ -55,35 +54,60 @@ private RuleEntryParams getRuleEntryParams(String commandId, List<Object> argume
...
+        JavaModelManager.getIndexManager().waitForIndex(true, monitor);
+        DeltaProcessingState state = JavaModelManager.getDeltaState();
+        logInfo("sourceAttachements: " + state.sourceAttachments);
+
+        Map<IPath, IPath> attachments = state.sourceAttachments;
+        Thread.sleep(5000);
+        while (attachments.size() != state.sourceAttachments.size()) {
+            logInfo("waiting size: " + state.sourceAttachments.size());
+            attachments = state.sourceAttachments;
+            Thread.sleep(10000);
```
Comment (pranavgaikwad) [inline java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java:87]: Just wondering if we know the ideal alternative solution for sleep()?

**Review by pranavgaikwad — APPROVED** [review]
just a question about sleep() call. it's not too insane but just wondering whether in an ideal world, we would be able to solve it with an alternative?

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java (hunk 1 above)
Comment (fabianvf) [reply java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java:87]: @pranavgaikwad no, I think it's going to be kind of tough unfortunately. We may want to toss some TODOs on those though so we don't forget.

**Review by fabianvf — APPROVED** [review]
TODOs on the sleeps would probably be good but other than that lgtm

**Comment by shawn-hurley** [comment]
There is probably some call somewhere that would allow us to do something with the monitor interface. Let's add an issue to track and a todo referencing that!

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java (hunk 1 above)
Comment (shawn-hurley) [reply java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java:89]: added todo and issue

**Pull request merged**